packages/mobx-react-lite/src/observer.ts

```diff
@@ -79,10 +77,6 @@ export function observer<P extends object, TRef = {}>(
     }
 
     // The working of observer is explained step by step in this talk: https://www.youtube.com/watch?v=cPF4iBedoF0&feature=youtu.be&t=1307
-    if (isUsingStaticRendering()) {
-        return baseComponent
```

Comment:
Is this change intended? Is that to make sure we can disable the flag as done below? I'm wondering if it would be simpler in that case to just wrap a sheep wrapper function around the original component definition so that all the observer setup stuff is skipped?

Reply:
Yes it's intended. Once you wrap it in a different function, you have to do almost the same machinery - forward ref, hoist name, statics, etc etc, maybe except memo ... basically most of the code here has nothing to do with setting up the "observer", it's dealing with the fact that we have to replace the original function with a new one. I could do the check a bit earlier, like eg here (avoiding `useObserver` + extra closure):
```
let observerComponent = (props: any, ref: React.Ref<TRef>) => {
  return useObserver(() => render(props, ref), baseComponentName)
}
```
but note `useObserver` is still part of the public API and can be used standalone.
Btw notice this is another place where proxy (instead of a wrapper) would make things easier.

One more thing to note - one solution to this problem is to simply say, don't use `observableRequiresReaction: true` on the server, it's useless anyway. The point of the PR is to provide something less obstructive.

Reply:
Ok, sounds good!